### PR TITLE
Feature/1854 OpenCL /prim signatures part 2

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR builds on #1859 and does the following changes to the /prim support in OpenCL:
- renames elewise_multiplication/elewise_division to elt_multiply/elt_divide
- adds crossprod, tcrossprod, fabs, log1m_inv_logit, logit, divide, trunc

## Tests

All new functions have tests.

## Side Effects

/

## Release notes

Renamed elewise_multiplication/elewise_division to elt_multiply/elt_divide in the kernel generator and added OpenCL support to /prim functions crossprod, tcrossprod, fabs, log1m_inv_logit, logit, divide

## Checklist

- [x] Math issue #1854

- [x] Copyright holder: Rok Češnovar

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
This is ready for review.

File: stan/math/opencl/prim/divide_columns.hpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ inline void divide_columns(const matrix_cl<T1>& A, const matrix_cl<T2>& B) {
  */
 template <typename T1, typename T2, typename = require_all_arithmetic_t<T1, T2>>
 inline void divide_columns(const matrix_cl<T1>& A, const T2& divisor) {
-  A = elewise_division(A, divisor);
+  A = elt_divide(A, divisor);
```
[the remark this answers is not in the document]
Comment (rok-cesnovar) [reply stan/math/opencl/prim/divide_columns.hpp:57]: Added to the list in https://github.com/stan-dev/math/issues/1854

File: stan/math/opencl/prim/crossprod.hpp [hunk 2]
```diff
@@ -19,8 +19,9 @@ namespace math {
 template <typename T_A,
           typename = require_all_kernel_expressions_and_none_scalar_t<T_A>>
 inline auto crossprod(T_A&& A) {
```
Comment (t4c1) [inline stan/math/opencl/prim/crossprod.hpp:21]: [optional]
```suggestion
inline matrix_cl<typename std::decay_t<T_A>::Scalar> crossprod(T_A&& A) {
```

File: stan/math/opencl/prim/tcrossprod.hpp [hunk 3]
```diff
@@ -19,8 +19,9 @@ namespace math {
 template <typename T_A,
           typename = require_all_kernel_expressions_and_none_scalar_t<T_A>>
 inline auto tcrossprod(T_A&& A) {
```
Comment (t4c1) [inline stan/math/opencl/prim/tcrossprod.hpp:21]: [optional]
```suggestion
inline matrix_cl<typename std::decay_t<T_A>::Scalar> tcrossprod(T_A&& A) {
```

**Pull request merged**